### ISSUE TITLE
DateTime.MinValue (default) is wrong probably

### ISSUE THREAD

**Issue by OlegZ-XC** [dotnet/runtime#47460]
next code throws exception when machine time (or region) in UTC+x but OK with negative UTC

using System;
class P { static void Main() {
    DateTimeOffset dto = DateTime.MinValue;
    Console.WriteLine(dto);
} }

.MinValue is default value so when u create some struct with defaults and then assign it to DateTimeOffset app behaviour is changed depends from user/machine region. imo its wrong.

simple workaround:
  DateTime.MinValue = new DateTime(0, Kind.Utc)

**Comment by huoyaoyuan** [comment]
Change the behavior treating `Unspecified` As `Local` is breaking.

> simple workaround for .NET:
> ` default(DateTime) = DateTime.MinValue = new DateTime(0, Kind.Utc)` // no exceptions with .Utc

No, this isn't possible. The `enum DateTimeKind` has value `Unspecified = 0`.


The simplest correct way is to use `default(DateTimeOffset)`. There should be an analyzer to warn on potential acquiring of `DateTimeKind.Unspecified`.

**Comment by OlegZ-XC** [comment]
```
    public DateTimeOffset(DateTime dateTime)
    {
      TimeSpan offset = dateTime.Kind == DateTimeKind.Utc ? new TimeSpan(0L) : TimeZoneInfo.GetLocalUtcOffset(dateTime, TimeZoneInfoOptions.NoThrowOnInvalidTime);
      this.m_offsetMinutes = DateTimeOffset.ValidateOffset(offset);
      this.m_dateTime = DateTimeOffset.ValidateDate(dateTime, offset);
    }
```
change to
```
      var useZeroOffset = dateTime == default(DateTime) ||  dateTime.Kind == DateTimeKind.Utc;
      TimeSpan offset = useZeroOffset ? new TimeSpan(0L) : TimeZoneInfo.GetLocalUtcOffset(dateTime, 

```

**Comment by huoyaoyuan** [comment]
Yes, changing the behavior in the conversion constructor is **possible**, but it's still breaking because it changes behavior.

The correct approach should be allowing 0 ticks with positive offset, because the internal representation of `DateTimeOffset` supports it.

**Comment by BrennanConroy** [comment]
FYI, ASP.NET Core just hit this in preview5 https://github.com/dotnet/aspnetcore/issues/33634

I understand that this is "by design", but there really shouldn't be different behavior for something like this based on time zones. Especially when the difference is between working or throwing.

**Comment by CuteLeon** [comment]
Met this issue https://github.com/dotnet/aspnetcore/issues/33634 when upgrading https://github.com/cuteleon/hacksystem from .Net 6 Preview 1 to .Net 6 Preview 5

**Comment by roklenardic** [comment]
Is there a workaround for #33634? JWT authentication is broken and this issue has been present since January...

**Comment by CuteLeon** [comment]
> Met this issue [dotnet/aspnetcore#33634](https://github.com/dotnet/aspnetcore/issues/33634) when upgrading https://github.com/cuteleon/hacksystem from .Net 6 Preview 1 to .Net 6 Preview 5

Issue gone in Preview 6 version.

**Comment by roklenardic** [comment]
I confirm that this is gone in Preview 6 as well.

**Comment by OlegZ-XC** [comment]
> Issue gone in Preview 6 version.

I can't check this before the stable Release, so if you are sure then close the issue.

**Comment by BrennanConroy** [comment]
I think they were talking about the linked JWT issue being gone in preview6... not the DateTimeOffset issue being fixed.

**Comment by CuteLeon** [comment]
Yeah actually,I mean JWT works in preview 6 currently,no more time-zone related exceptions thrown anymore.
